prog/prog.go

```diff
@@ -12,6 +12,9 @@ type Prog struct {
 	Target   *Target
 	Calls    []*Call
 	Comments []string
+
+	// Was deserialized using Unsafe mode, so can do unsafe things.
+	isUnsafe bool
```

Comment:
I worry that such an extra Prog field may bring us more headache than it solves. 

If some validation logic depends on `isUnsafe`, then we have to ensure that the resulting "conditionally valid" parts of the program never ever leak to Progs with `isUnsafe=false`:
* We must keep the flag after prog cloning. And we don't always copy the program by an explicit Clone() -- there are cases when we explicitly create a `&Prog{}` object and fill its `Calls` field.
* The conditionally valid parts of the program must never reach corpus and in general be used during fuzzing.

Yes, now we only actually use the feature for machine checks, and we don't clone such programs, so we evidently don't hit problems, but it looks fragile. Will we remember all the subtleties in a few years from now?

I think we need to either:
* Add more `if p.isUnsafe { panic() }` assertions throughout the fuzzer code. Or
* Solve the problem differently.

What do you think about explicitly splitting the validation into two parts:
1) The overall consistency that we must always check after deserialization / after any modifications of the Prog representation.
2) The checks that ensure that the fuzzer does not mangle the environment/the executor process too much. It's the sanitization and it can also be the address range checks.

We only need (2) for the outputs of `pkg/fuzzer` (+ for the corpus and the syz-hub progs, which also go through `pkg/fuzzer` anyway). So, maybe, we should export some `prog.Sanitize()` method and just invoke it for the progs that `fuzzer.NextInput()` returned?

Then we can execute whatever we want when we test the instance.

Reply:
> So, maybe, we should export some prog.Sanitize() method and just invoke it for the progs that fuzzer.NextInput() returned (e.g. as https://github.com/google/syzkaller/pull/4762)?

We don't run these checks in production. We need to run them in tests, preferably for each program produced.

We have very few entry points into the prog package, the ones that produce new programs are Clone/Generate/Mutate.
I would try to restrict use of unsafe programs for now, and see if they really cause any issues in future.